relay-server/src/services/processor/profile.rs

```diff
@@ -95,6 +97,54 @@ pub fn transfer_id(event: &mut Annotated<Event>, profile_id: Option<ProfileId>)
     }
 }
 
+/// Removes the profile context from the transaction item if there is an active rate limit.
+///
+/// With continuous profiling profile chunks are ingested separately to transactions,
+/// in the case where these profiles are rate limited the link on the associated transaction(s)
+/// should also be removed.
+///
+/// See also: <https://github.com/getsentry/relay/issues/5071>.
+pub fn remove_context_if_rate_limited(
+    event: &mut Annotated<Event>,
+    envelope: &ManagedEnvelope,
+    ctx: Context<'_>,
+) {
+    let Some(event) = event.value_mut() else {
+        return;
+    };
+
+    // There is always only either a transaction profile or a continuous profile, never both.
+    //
+    // If the `profiler_id` is set on the context, it is for a continuous profile, the case we want
+    // to handle here.
+    // If it is empty -> do nothing.
+    let profile_ctx = event.context::<ProfileContext>();
+    if profile_ctx.is_none_or(|pctx| pctx.profiler_id.is_empty() || !pctx.profile_id.is_empty()) {
+        return;
+    }
```

Comment:
I think you can be more strict here because we're specifically targeting the case when it's a continuous/ui profile.

```suggestion
    if profile_ctx.is_none_or(|pctx| !pctx.profiler_id.is_empty()) {
        return;
    }
```